pkg/utils/ocmWrapper.go

```diff
@@ -0,0 +1,228 @@
+	connection, err := ocm.NewConnection().Build()
+	if err != nil {
+		return "", "", fmt.Errorf("failed to create OCM connection: %v", err)
+	}
+	defer connection.Close()
+	// Get the client for the resource that manages the collection of clusters:
+	clusterCollection := connection.ClustersMgmt().V1().Clusters()
+	clusters, err := getClusters(clusterCollection, clusterKey)
+	if err != nil {
+		return "", "", fmt.Errorf("failed to get cluster '%s': %v", clusterKey, err)
+	}
+
+	if len(clusters) == 1 {
+		for _, v := range clusters {
+			clusterId = v.ID()
+			clusterName = v.Name()
+		}
+	} else {
+		cluster, err := doSurvey(clusters)
+		if err != nil {
+			return "", "", fmt.Errorf("can't find cluster: %v", err)
+		}
+		clusterId = cluster.ID()
+		clusterName = cluster.Name()
+	}
+	return clusterId, clusterName, nil
+}
+
+func (*DefaultOCMInterfaceImpl) GetOCMAccessToken() (*string, error) {
+	// Get ocm access token
+	logger.Debugln("Finding ocm token")
+	connection, err := ocm.NewConnection().Build()
+	if err != nil {
+		return nil, fmt.Errorf("failed to create OCM connection: %v", err)
+	}
+	defer connection.Close()
+	accessToken, _, err := connection.Tokens()
+	if err != nil {
+		return nil, err
+	}
+
+	logger.Debugln("Found OCM access token")
+	accessToken = strings.TrimSuffix(accessToken, "\n")
+
+	return &accessToken, nil
+}
+
+// GetClusterInfoByID calls the OCM to retrieve the cluster info
+// for a given internal cluster id.
+func (*DefaultOCMInterfaceImpl) GetClusterInfoByID(clusterId string) (*cmv1.Cluster, error) {
+	// Create the client for the OCM API:
+	connection, err := ocm.NewConnection().Build()
+	if err != nil {
+		return nil, fmt.Errorf("failed to create OCM connection: %v", err)
+	}
+	defer connection.Close()
+
+	// Get the cluster info based on the cluster id
+	response, err := connection.ClustersMgmt().V1().Clusters().Cluster(clusterId).Get().Send()
+	if err != nil {
+		return nil, fmt.Errorf("can't retrieve cluster for id '%s': %v", clusterId, err)
+	}
+	cluster, ok := response.GetBody()
+	if !ok {
+		return nil, fmt.Errorf("can't retrieve cluster for id '%s', nil value", clusterId)
+	}
+	return cluster, nil
+}
+
+// IsProduction checks if OCM is currently in production env
+func (*DefaultOCMInterfaceImpl) IsProduction() (bool, error) {
+	// Create the client for the OCM API:
+	connection, err := ocm.NewConnection().Build()
+	if err != nil {
+		return false, fmt.Errorf("failed to create OCM connection: %v", err)
+	}
+	defer connection.Close()
+
+	return connection.URL() == "https://api.openshift.com", nil
+}
+
+// GetBackplaneURL from local settings
+func (*DefaultOCMInterfaceImpl) GetBackplaneURL() (string, error) {
+
+	// get backplane URL from BACKPLANE_URL env variables
+	bpURL, hasURL := os.LookupEnv(info.BACKPLANE_URL_ENV_NAME)
+	if hasURL {
+		if bpURL != "" {
+			return bpURL, nil
+		} else {
+			return "", fmt.Errorf("%s env variable is empty", info.BACKPLANE_URL_ENV_NAME)
+		}
+	} else {
```

Comment:
@supreeth7 pls, add [this unit test ](https://github.com/openshift/backplane-cli/blob/c7572c1f17b1a9f84146f673187fb5f42e1fc417/pkg/utils/ocmWrapper_test.go) for covering the BP_URL setting via env vars